lib/puma/cluster.rb

```diff
@@ -322,6 +343,34 @@ def setup_signals
               w.last_status[:requests_count] >= fork_requests
           end
         end
+
+      end
+      if @options[:mold_worker]
+        Signal.trap "SIGURG" do
+          mold_and_refork!
+        end
+        current_interval_index = 0
+        Signal.trap "SIGUSR1" do
```

Comment:
on USR1, go back to the original state (replace all workers and molds with fresh forks of the cluster parent and restart the interval series)